### ISSUE TITLE
Backport Layer 4 Load Balancing Disabled With MetalLB Configured

### ISSUE THREAD

**Issue by alena1108** [rancher/rancher#20112]
Backport https://github.com/rancher/rancher/issues/18275

**Comment by alena1108** [comment]
Can be tested on rc8

**Comment by sangeethah** [comment]
Able to reproduce the issue when testing with v2.2.2 where RKE cluster (with no cloud provider set) , has `Layer4 - LoadBalancer ` option disabled in `Port Mapping` section when deploying workloads.

<img width="551" alt="Screen Shot 2019-05-10 at 5 17 32 PM" src="https://user-images.githubusercontent.com/4266958/57562475-16457100-7348-11e9-9389-3f0df12a9381.png">

In cluster api, `"enabled": false,` is set for loadBalancerCapabilities.


```
"loadBalancerCapabilities": {
"enabled": false,
"healthCheckSupported": false,
"type": "/v3/schemas/loadBalancerCapabilities"
},
"nodePoolScalingSupported": true,
"nodePortRange": "30000-32767",
"type": "/v3/schemas/capabilities"
},
```

Steps followed to validate the fix in v2.2.3-rc8 :

1. Create a DO cluster (With no cloud provider option set).
2. Deploy a workload. Notice that  `Layer4 - LoadBalancer ` option is enabled in `Port Mapping` section when deploying workloads.

<img width="1017" alt="Screen Shot 2019-05-10 at 5 18 08 PM" src="https://user-images.githubusercontent.com/4266958/57562554-af748780-7348-11e9-95c4-e0272d717547.png">
In cluster api , there is no `enabled`  flag set for loadBalancerCapabilities.

```
"loadBalancerCapabilities": {
"healthCheckSupported": false,
"type": "/v3/schemas/loadBalancerCapabilities"
},
"nodePoolScalingSupported": true,
"nodePortRange": "30000-32767",
"type": "/v3/schemas/capabilities"
},
```
